app/models/course.rb

```diff
@@ -18,6 +18,19 @@ class Course < ApplicationRecord
   has_associated_audits
   audited
 
+  attribute :subordinate_subject_id, :integer
+  after_initialize do
+    if is_primary?
+      self.master_subject_id ||= course_subjects.first&.subject_id
+    else
+      sub = course_subjects.select(&:position)
+
+      self.master_subject_id ||= sub.first&.subject&.id
+
+      self.subordinate_subject_id ||= sub&.second&.subject&.id
+    end
```

Comment:
This does not account for Further Education, but there is no form inputs for further Education level courses. The `master` and `subordinate` properties are only used for input and validation.